bundler/lib/bundler/dsl.rb

```diff
@@ -278,8 +278,17 @@ def add_git_sources
         warn_deprecated_git_source(:github, <<-'RUBY'.strip, 'Change any "reponame" :github sources to "username/reponame".')
 "https://github.com/#{repo_name}.git"
         RUBY
-        repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
-        "https://github.com/#{repo_name}.git"
+        if repo_name =~ %r{\Ahttps://github\.com/(\w+/\w+)/pull/(\d+)\z}
```

Comment:
Should we put this to constant and freeze?

Reply:
Regexps are frozen by default https://github.com/ruby/ruby/pull/2705

Reply:
Sure, but that's not part of all supported Ruby versions. I'm just trying to stick with already used code patterns, as we do this (put regexp into constant and freeze) at various places.

Reply:
I think it's fine as is @simi. We sometimes do that, but sometimes we don't, I don't think there's a clear existing pattern.

Reply:
Up to you, I edited, but I'd suggest you revise that pattern, if some code was trying to mutate a regexp literal it would fail on newer Ruby CI, that should be enough to prevent this error.

Reply:
I see @casperisfine already updated it, so that's fine too 😄.